### ISSUE TITLE
Cannot load the graph: cluster [unknown] is not found or is not accessible for Kiali

### ISSUE THREAD

**Issue by dtwilliamsWork** [kiali/kiali#7672]
### Describe the bug

We have a fresh install of Kiali v1.88.0 using the Operator and CR but the graphs will not load.

We have Istio v1.23 installed in Ambient mode with the Kubernetes Gateway API instead of the Istio Ingress and Egress gateways. Istio base, cni, istiod and ztunnel are all installed using the Helm chart with default values, apart from setting profile=ambient.

```
kc -n istio-system logs deployment.apps/kiali
2024/08/27 13:52:27 maxprocs: Leaving GOMAXPROCS=4: CPU quota undefined
2024-08-27T13:52:27Z INF Kiali: Version: v1.88.0, Commit: 8b2946fee5f39c6f26fde4599a274b80cc75d04b, Go: 1.22.4
2024-08-27T13:52:27Z INF Using authentication strategy [openid]
2024-08-27T13:52:27Z INF Some validation errors will be ignored [KIA1301]. If these errors do occur, they will still be logged. If you think the validation errors you see are incorrect, please report them to the Kiali team if you have not done so already and provide the details of your scenario. This will keep Kiali validations strong for the whole community.
2024-08-27T13:52:27Z INF Initializing Kiali Cache
2024-08-27T13:52:27Z INF [Kiali Cache] Waiting for cluster-scoped cache to sync
2024-08-27T13:52:27Z INF [Kiali Cache] Started
2024-08-27T13:52:27Z INF [Kiali Cache] Kube cache is active for cluster: [Kubernetes]
2024-08-27T13:52:27Z INF [Prom Cache] Enabled
2024-08-27T13:52:27Z INF Server endpoint will start at [:20001/]
2024-08-27T13:52:27Z INF Server endpoint will serve static content from [/opt/kiali/console]
2024-08-27T13:52:27Z INF Starting Metrics Server on [:9090]
2024-08-27T13:52:27Z INF Create jaeger GRPC client jaeger-instance-query.jaeger:16685
2024-08-27T13:56:12Z WRN Could not read the session: session not found: cookie kiali-token-Kubernetes does not exist in request
2024-08-27T13:56:13Z WRN No Grafana dashboard found for pattern 'Istio Control Plane Dashboard'
2024-08-27T13:56:13Z WRN No Grafana dashboard found for pattern 'Istio Wasm Extension Dashboard'
2024-08-27T13:56:14Z WRN Not all required K8s Gateway API CRDs are installed for version: gateway.networking.k8s.io/v1alpha2, expected: TCPRoute, TLSRoute
2024-08-27T13:56:29Z ERR cluster [unknown] is not found or is not accessible for Kiali: goroutine 1596 [running]:
runtime/debug.Stack()
	/opt/hostedtoolcache/go/1.22.4/x64/src/runtime/debug/stack.go:24 +0x5e
github.com/kiali/kiali/handlers.handlePanic({0x26fa7d0, 0xc001237a88})
	/home/runner/work/kiali/kiali/handlers/graph.go:117 +0x1ad
panic({0x1e76b60?, 0xc000cb2560?})
	/opt/hostedtoolcache/go/1.22.4/x64/src/runtime/panic.go:770 +0x132
github.com/kiali/kiali/graph.CheckError(...)
	/home/runner/work/kiali/kiali/graph/util.go:38
github.com/kiali/kiali/graph/telemetry/istio/appender.addBadging(0xc0023fe9f0, 0xc0023fe990, 0xc0025766c0)
	/home/runner/work/kiali/kiali/graph/telemetry/istio/appender/istio_details.go:63 +0x445
github.com/kiali/kiali/graph/telemetry/istio/appender.IstioAppender.AppendGraph({0x23834c1?}, 0xc0023fe9f0, 0xc0023fe990, 0xc0025766c0)
	/home/runner/work/kiali/kiali/graph/telemetry/istio/appender/istio_details.go:48 +0x73
github.com/kiali/kiali/graph/telemetry/istio.BuildNamespacesTrafficMap({_, _}, {0xc0023fe870, {0x0, {0xc0005e0ee0, 0x7, 0x7}}, 0x0, 0x1, 0xc0039cb170, ...}, ...)
	/home/runner/work/kiali/kiali/graph/telemetry/istio/istio.go:80 +0x674
github.com/kiali/kiali/graph/api.graphNamespacesIstio({_, _}, _, _, {{0x2389047, 0x9}, {0x23834c1, 0x5}, {{0xc0006b6753, 0x15}, ...}, ...})
	/home/runner/work/kiali/kiali/graph/api/api.go:52 +0xf0
github.com/kiali/kiali/graph/api.GraphNamespaces({_, _}, _, {{0x2389047, 0x9}, {0x23834c1, 0x5}, {{0xc0006b6753, 0x15}, {0xdf8475800, ...}}, ...})
	/home/runner/work/kiali/kiali/graph/api/api.go:33 +0x2f3
github.com/kiali/kiali/routing.NewRoutes.GraphNamespaces.func14({0x26fa7d0, 0xc001237a88}, 0xc0043ae7e0)
	/home/runner/work/kiali/kiali/handlers/graph.go:70 +0x1c5
net/http.HandlerFunc.ServeHTTP(0x23932b8?, {0x26fa7d0?, 0xc001237a88?}, 0x120?)
	/opt/hostedtoolcache/go/1.22.4/x64/src/net/http/server.go:2166 +0x29
github.com/kiali/kiali/routing.NewRouter.metricHandler.func6({0x26fa740, 0xc0005e0e70}, 0xc0043ae7e0)
	/home/runner/work/kiali/kiali/routing/router.go:290 +0x11f
net/http.HandlerFunc.ServeHTTP(0xc0043ae240?, {0x26fa740?, 0xc0005e0e70?}, 0xc0039caea0?)
	/opt/hostedtoolcache/go/1.22.4/x64/src/net/http/server.go:2166 +0x29
github.com/kiali/kiali/routing.NewRouter.(*AuthenticationHandler).Handle.func7({0x26fa740, 0xc0005e0e70}, 0xc0043ae240)
	/home/runner/work/kiali/kiali/handlers/authentication.go:104 +0x3ab
net/http.HandlerFunc.ServeHTTP(0x2707db8?, {0x26fa740?, 0xc0005e0e70?}, 0x26d2a40?)
	/opt/hostedtoolcache/go/1.22.4/x64/src/net/http/server.go:2166 +0x29
github.com/kiali/kiali/server.plainHttpMiddleware.func1({0x26fa740?, 0xc0005e0e70?}, 0xc0039cab70?)
	/home/runner/work/kiali/kiali/server/server.go:189 +0x5c
net/http.HandlerFunc.ServeHTTP(0xc0043ae120?, {0x26fa740?, 0xc0005e0e70?}, 0x41a1d8?)
	/opt/hostedtoolcache/go/1.22.4/x64/src/net/http/server.go:2166 +0x29
github.com/gorilla/mux.(*Router).ServeHTTP(0xc001019380, {0x26fa740, 0xc0005e0e70}, 0xc0043ae000)
	/home/runner/go/pkg/mod/github.com/gorilla/mux@v1.8.1/mux.go:212 +0x1e2
github.com/NYTimes/gziphandler.GzipHandlerWithOpts.func1.1({0x26f3d80, 0xc003a9d0a0}, 0xc0043ae000)
	/home/runner/go/pkg/mod/github.com/!n!y!times/gziphandler@v1.1.1/gzip.go:336 +0x27e
net/http.HandlerFunc.ServeHTTP(0xc0027ebba0?, {0x26f3d80?, 0xc003a9d0a0?}, 0x7aa7da?)
	/opt/hostedtoolcache/go/1.22.4/x64/src/net/http/server.go:2166 +0x29
net/http.(*ServeMux).ServeHTTP(0x471559?, {0x26f3d80, 0xc003a9d0a0}, 0xc0043ae000)
	/opt/hostedtoolcache/go/1.22.4/x64/src/net/http/server.go:2683 +0x1ad
net/http.serverHandler.ServeHTTP({0xc003e19950?}, {0x26f3d80?, 0xc003a9d0a0?}, 0x6?)
	/opt/hostedtoolcache/go/1.22.4/x64/src/net/http/server.go:3137 +0x8e
net/http.(*conn).serve(0xc003e64510, {0x2707db8, 0xc0020bb110})
	/opt/hostedtoolcache/go/1.22.4/x64/src/net/http/server.go:2039 +0x5e8
created by net/http.(*Server).Serve in goroutine 352
	/opt/hostedtoolcache/go/1.22.4/x64/src/net/http/server.go:3285 +0x4b4
```

Our CR is

```
apiVersion: kiali.io/v1alpha1
kind: Kiali
metadata:
  name: kiali
  namespace: istio-system
spec:
  istio_namespace: istio-system
  istio_labels:
    app_label_name: "app.kubernetes.io/name"
    version_label_name: "app.kubernetes.io/version"

  auth:
    # strategy: "anonymous"
    strategy: openid
    openid:
      client_id: "xxx"
      issuer_uri: "xxx"
      authorization_endpoint: "xxx"
      scopes: ["openid", "email"]
      disable_rbac: true
      insecure_skip_verify_tls: false
      username_claim: "email"

  deployment:
    service_type: NodePort
    # accessible_namespaces: ["bookinfo"]
    view_only_mode: false

  server:
    web_root: "/"
    web_fqdn: ${var.kiali_domain}
    web_schema: https

  external_services:
    prometheus:
      url: "http://kps-prometheus.monitoring:9090/"
    grafana:
      enabled: true
      auth:
        ca_file: ""
        insecure_skip_verify: true
        password: ${var.infra_secrets["GRAFANA_ADMIN_PASSWORD_BLUE"]}
        token: ""
        type: basic
        use_kiali_token: false
        username: "admin"
      in_cluster_url: 'http://kps-grafana.monitoring:443/'
      url: 'https://${var.grafana_domain}'
      dashboards:
      - name: "Istio Service Dashboard"
        variables:
          namespace: "var-namespace"
          service: "var-service"
      - name: "Istio Workload Dashboard"
        variables:
          namespace: "var-namespace"
          workload: "var-workload"
      - name: "Istio Mesh Dashboard"
      - name: "Istio Control Plane Dashboard"
      - name: "Istio Performance Dashboard"
      - name: "Istio Wasm Extension Dashboard"
    tracing:
      enabled: true
      in_cluster_url: 'http://jaeger-instance-query.jaeger:16685/jaeger'
      use_grpc: true
      url: 'https://${var.jaeger_domain}/'
```

### Expected Behavior

Graphs load correctly.

### Environment

- **Kiali version: v1.88
- **Istio version: 1.23.0
- **Kubernetes version: v1.30.3-eks-2f46c53
- **Other notable environmental factors:**
-
<img width="271" alt="Screenshot 2024-08-27 at 15 00 09" src="https://github.com/user-attachments/assets/f7badb5f-f0d1-479c-9cd9-24b89d088713">

**Comment by jmazzitelli** [comment]
Just to confirm, Istio is not installed in some kind of multi-cluster environment, correct? A single cluster with a single control plane?

**Comment by dtwilliamsWork** [comment]
correct, single EKS cluster

**Comment by josunect** [comment]
@dtwilliamsWork could you check what is the value in `destination_cluster` and `source_cluster` for the namespace that you are trying to see the graph, in Prometheus for the metric `istio_tcp_received_bytes_total{}`?

When the `cluster_name` is not set in Kiali, Kiali will try to look for the value `CLUSTER_ID` in the Istio deployment environment var. When Kial is not able to access to this value, it will set the `cluster_name` to `Kubernetes` as a default. I guess there is some mismatch there.

**Comment by jshaughn** [comment]
It's unclear why you're seeing an "unknown" cluster value, but regardless of why, we will need to protect against this in the code, so let me take a look at adding that protection.

**Comment by jshaughn** [comment]
This fix will need to be backported to v1.89.
